modules/clients.py

```diff
@@ -27,51 +27,43 @@ def done_at(self, time):
-        self.process_time = process_time
-        self.queue = simpy.Store(env)  # the queue of requests
+        self.sleep_time = sleep_time
+        self.queue = simpy.Store(env)               # the queue of requests
         self.action = env.process(self.send_request())
-        self.create_request = env.process(self.create_request(create_request_rate, max_requests))
+        self.create_request = env.process(self.create_requests(create_request_time, max_requests, request_exec_time, memory))
 
     def send_request(self):
         while True:
-            if len(self.queue.items) > 0:  # check if there is any request to be processed
-                request = yield self.queue.get()  # get a request from store
-                print("At %.3f, CLIENTS a client sent a request" % self.env.now)
+            if len(self.queue.items) > 0:           # check if there is any request to be processed
+                request = yield self.queue.get()    # get a request from store
                 yield self.env.process(self.server.request_arrived(request))
 
-            yield self.env.timeout(self.process_time) # wait for...
+            yield self.env.timeout(self.sleep_time) # wait for...
 
-    def create_request(self, create_request_rate, max_requests):
-        """ Create requests """
-        count = 1
-        while count <= max_requests:
-            duration = 0.035
-            memory = 0.02
-            request = Request(self.env.now, duration, memory, self)
-            print("At %.3f, CLIENTS a client made a request" % self.env.now)
+    def create_requests(self, create_request_time, max_requests, request_exec_time, memory):
+        count_requests = 1
+        while count_requests <= max_requests:
+            request = Request(self.env.now, request_exec_time, memory, self)
             yield self.queue.put(request)
-            yield self.env.timeout(create_request_rate)
-            count += 1
+            yield self.env.timeout(create_request_time)
+            count_requests += 1
 
     def successfully_sent(self, request):
-        print("At %.3f, CLIENTS request successfully sent" % self.env.now)
         request.sent_at(self.env.now)
         yield self.env.timeout(0)
 
     def sucess_request(self, request):
-        print("At %.3f, CLIENTS request successful attended" % self.env.now)
-        print("At %.3f, CLIENTS Service time: %.3f" % (self.env.now, self.env.now - request._sent_time))
         request.done_at(self.env.now)
         self.requests.append(request)
         yield self.env.timeout(0)
 
-    def refused_request(self, request, unavailable_until):
-        print("At %.3f, CLIENTS refused request.Server is no available until %.3f" % (self.env.now, unavailable_until))
+    def shed_request(self, request, unavailable_until):
         # don't let the client send new requests until unavailable_until
-        # put the request in the front of the queue
+        # put the request in the front of the queue?
```

Comment:
No. The server must send the request back to the client (possibly a load balancer) which will redirect it to another server.

Reply:
The queue quoted is the clients queue. The next version of our simulator will have a load balancer and it will satisfy that aspect. By the way, we plan insert this load balance between the server and clients.

Reply:
Cool.